jupyterlab_gcpscheduler/src/components/dialog.tsx

```diff
@@ -263,11 +250,20 @@ export class SchedulerDialog extends React.Component<Props, State> {
   }
 
   // Casts to GcpSettings shape from JSONObject
-  private _settingsChanged(newSettings: ISettingRegistry.ISettings) {
+  private async _settingsChanged(newSettings: ISettingRegistry.ISettings) {
     const settings = (newSettings.composite as unknown) as GcpSettings;
     if (settings.projectId) {
       console.info(`Using ${settings.projectId} for GCP API calls`);
       this.props.projectStateService.projectId = settings.projectId;
+    } else {
+      let projectId = undefined;
+      try {
+        projectId = await this.props.gcpService.projectId;
+      } catch (err) {
+        console.info(`${err} occurred when trying to get projectId`);
+      }
+      this.props.projectStateService.projectId = projectId;
+      settings.projectId = projectId;
```

Comment:
Per our Chat conservation, if we want to continue to persist the projectId in the settings, we should add a call to `this.props.settings.set('projectId'...` similar to https://github.com/GoogleCloudPlatform/jupyter-extensions/blob/e534ba1bbb80b332419210af4421d3979db94653/jupyterlab_gcpscheduler/src/components/dialog.tsx#L298.

Reply:
Done